doc/en/assert.rst

```diff
@@ -246,8 +246,7 @@ the conftest file::
            f1 = Foo(1)
            f2 = Foo(2)
    >       assert f1 == f2
-   E       assert Comparing Foo instances:
-   E            vals: 1 != 2
+   E       AssertionError
```

Comment:
I'll let @nicoddemus decide - I think we should probably fix #1784 before releasing 3.0, right? If so, we'd need to re-run regendoc anyways, so not much point in merging then from my POV.